### ISSUE TITLE
DynamoDB tables w/ Streams configured no longer work without kinesis.

### ISSUE THREAD

**Issue by workmanw** [localstack/localstack#3120]
# Type of request: This is a ...

[X] bug report
[ ] feature request

# Detailed description

Using DynamoDB with Streams seems to require Kinesis in 0.12.0. It did not in 0.11.6.

## Expected behavior

Expect that DDB can be used w/ Streams independently of Kinesis.

## Actual behavior

```
2020-10-16T22:36:14:WARNING:localstack.utils.server.http2_server: Error in proxy handler for request POST http://ad04a495fcaf:4566/: Unable to find listener for service "kinesis" - please make sure to include it in $SERVICES Traceback (most recent call last):
  File "/opt/code/localstack/localstack/utils/server/http2_server.py", line 94, in index
    raise result
  File "/opt/code/localstack/localstack/utils/bootstrap.py", line 533, in run
    result = self.func(self.params)
  File "/opt/code/localstack/localstack/utils/async_utils.py", line 28, in _run
    return fn(*args, **kwargs)
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 560, in handler
    response = modify_and_forward(method=method, path=path_with_params, data_bytes=data, headers=headers,
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 333, in modify_and_forward
    listener_result = listener.forward_request(method=method,
  File "/opt/code/localstack/localstack/services/edge.py", line 81, in forward_request
    return do_forward_request(api, port, method, path, data, headers)
  File "/opt/code/localstack/localstack/services/edge.py", line 86, in do_forward_request
    result = do_forward_request_inmem(api, port, method, path, data, headers)
  File "/opt/code/localstack/localstack/services/edge.py", line 100, in do_forward_request_inmem
    raise HTTPErrorResponse(message, code=400)
localstack.utils.server.http2_server.HTTPErrorResponse: Unable to find listener for service "kinesis" - please make sure to include it in $SERVICES

2020-10-16T22:36:19:WARNING:bootstrap.py: Thread run method <function AdaptiveThreadPool.submit.<locals>._run at 0x7f5d46976a60>(None) failed: An error occurred (400) when calling the PutRecord operation:  Traceback (most recent call last):
  File "/opt/code/localstack/localstack/utils/bootstrap.py", line 533, in run
    result = self.func(self.params)
  File "/opt/code/localstack/localstack/utils/async_utils.py", line 28, in _run
    return fn(*args, **kwargs)
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 560, in handler
    response = modify_and_forward(method=method, path=path_with_params, data_bytes=data, headers=headers,
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 333, in modify_and_forward
    listener_result = listener.forward_request(method=method,
  File "/opt/code/localstack/localstack/services/edge.py", line 81, in forward_request
    return do_forward_request(api, port, method, path, data, headers)
  File "/opt/code/localstack/localstack/services/edge.py", line 86, in do_forward_request
    result = do_forward_request_inmem(api, port, method, path, data, headers)
  File "/opt/code/localstack/localstack/services/edge.py", line 106, in do_forward_request_inmem
    response = modify_and_forward(method=method, path=path, data_bytes=data, headers=headers,
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 401, in modify_and_forward
    updated_response = update_listener.return_response(**kwargs)
  File "/opt/code/localstack/localstack/services/dynamodb/dynamodb_listener.py", line 383, in return_response
    forward_to_ddb_stream(records)
  File "/opt/code/localstack/localstack/services/dynamodb/dynamodb_listener.py", line 654, in forward_to_ddb_stream
    dynamodbstreams_api.forward_events(records)
  File "/opt/code/localstack/localstack/services/dynamodbstreams/dynamodbstreams_api.py", line 55, in forward_events
    kinesis.put_record(StreamName=stream_name, Data=json.dumps(record), PartitionKey='TODO')
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 357, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 676, in _make_api_call
    raise error_class(parsed_response, operation_name)
```

# Steps to reproduce

Simple:

```
$ docker run -t -i -e SERVICES=dynamodb -p 4566:4566 localstack/localstack:0.12.0

$ aws dynamodb create-table \
  --table-name LocalstackTest \
  --attribute-definitions AttributeName=ID,AttributeType=S \
  --key-schema AttributeName=ID,KeyType=HASH \
  --provisioned-throughput ReadCapacityUnits=5,WriteCapacityUnits=5 \
  --stream-specification StreamEnabled=TRUE,StreamViewType=NEW_AND_OLD_IMAGES \
  --endpoint-url=http://localhost:4566

$ aws dynamodb put-item \
  --table-name LocalstackTest \
  --item "{ \"ID\": { \"S\": \"1\" } }" \
  --endpoint-url=http://localhost:4566


An error occurred (500) when calling the PutItem operation (reached max retries: 2):
```

**Comment by workmanw** [comment]
It's worth noting that if you enable kinesis (SERVICES=dynamodb,kinesis) this works as expected. Also if you go back to `0.11.6` this works as expected.

**Comment by whummer** [comment]
Thanks for reporting @workmanw . This should be fixed in #3123 - please report here if the problem persists with the `latest` version. Thanks

**Comment by workmanw** [comment]
@whummer I've confirmed this fix works with the test provided in this issue as well as two of our microservice's test suites. Thanks for the quick turn around, we really appreciate it!
